File: src/main/java/com/koldyr/csv/db/ConnectionsFactory.java
package com.koldyr.csv.db;

import java.sql.Connection;
import java.sql.DriverManager;

import org.apache.commons.pool2.BaseKeyedPooledObjectFactory;
import org.apache.commons.pool2.PooledObject;
import org.apache.commons.pool2.impl.DefaultPooledObject;

import com.koldyr.csv.model.ConnectionData;
import com.koldyr.csv.model.PoolType;

/**
 * Description of class ConnectionsFactory
 *
 * @created: 2018.03.09
 */
public class ConnectionsFactory extends BaseKeyedPooledObjectFactory<PoolType, Connection> {

    private final ConnectionData srcConfig;
    private final ConnectionData dstConfig;

    public ConnectionsFactory(ConnectionData srcConfig, ConnectionData dstConfig) {
        super();
        this.srcConfig = srcConfig;
        this.dstConfig = dstConfig;
    }

    @Override
    public Connection create(PoolType key) throws Exception {
        if (key == PoolType.SOURCE) {
            Connection connection = DriverManager.getConnection(srcConfig.getUrl(), srcConfig.getUser(), srcConfig.getPassword());
            connection.setAutoCommit(false);
            return connection;
        }

        Connection connection = DriverManager.getConnection(dstConfig.getUrl(), dstConfig.getUser(), dstConfig.getPassword());
        connection.setAutoCommit(false);
        return connection;
    }

    @Override
    public PooledObject<Connection> wrap(Connection obj) {
        return new DefaultPooledObject<>(obj);
    }

    @Override
    public void destroyObject(PoolType key, PooledObject<Connection> p) throws Exception {
        if (p != null && p.getObject() != null) {
            p.getObject().close();
        }
    }
}

File: src/main/kotlin/com/koldyr/csv/db/ConnectionsFactory.kt
package com.koldyr.csv.db

import java.sql.Connection
import java.sql.DriverManager
import org.apache.commons.pool2.BaseKeyedPooledObjectFactory
import org.apache.commons.pool2.PooledObject
import org.apache.commons.pool2.impl.DefaultPooledObject
import com.koldyr.csv.model.ConnectionData
import com.koldyr.csv.model.PoolType

/**
 * Description of class ConnectionsFactory
 *
 * @created: 2018.03.09
 */
class ConnectionsFactory(
    private val srcConfig: ConnectionData,
    private val dstConfig: ConnectionData
) : BaseKeyedPooledObjectFactory<PoolType, Connection>() {

    @Throws(Exception::class)
    override fun create(key: PoolType): Connection {
        if (key === PoolType.SOURCE) {
            val connection = DriverManager.getConnection(srcConfig.url, srcConfig.user, srcConfig.password)
            connection.autoCommit = false
            return connection
        }

        val connection = DriverManager.getConnection(dstConfig.url, dstConfig.user, dstConfig.password)
        connection.autoCommit = false
        return connection
    }

    override fun wrap(obj: Connection): PooledObject<Connection> = DefaultPooledObject(obj)

    @Throws(Exception::class)
    override fun destroyObject(key: PoolType, p: PooledObject<Connection>) {
        p.getObject()?.close()
    }
}
